### PR TITLE
chore(dataCleaner): flow

### PR DESCRIPTION
**Pull request by klerikdust**

### CODE REVIEW

**Comment by Moore2021** [comment]
need to complete this system and decide if we want it

**Comment by Moore2021** [comment]
Will need to add column on prod still

File: src/commands/developer/userdata.js [hunk 1]
```diff
@@ -296,6 +296,12 @@ module.exports = {
     async removeUserdata(client, reply, message, argu, locale) {
         const userId = argu[1]
         const sql = `DELETE FROM users WHERE user_id=$userID`
+        const updateConfig = `UPDATE guild_configurations SET setByUserId=$bot, updated_at = CURRENT_TIMESTAMP WHERE setByUserId=$userID`
```
Comment (klerikdust) [inline src/commands/developer/userdata.js:299]: I didn't quite understand this part😂
What does guild config has to do with user data deletion .. is it marking that there is an activity from the said user?

Comment (Moore2021) [reply src/commands/developer/userdata.js:299]: User id is tied to that data so if the user is deleted, the id needs to be updated to not break configurations

File: src/events/ready.js [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
  * Ready event.
  * @param {Client} annie Current bot/worker instance.
  * @return {void}
  */
 module.exports = function ready(annie) {
     annie.db.initializeDb()
+    const cleaner = new dataCleaner(annie)
+    let current_shard = (annie.guilds.cache.first()).shard.id
+    let last_shard = (annie.shard.ids)[(annie.shard.ids).length - 1]
+    if (current_shard == last_shard) {
```
[the remark this answers is not in the document]
Comment (Moore2021) [reply src/events/ready.js:14]: i only wanted it to run on the last shard, we have roughly 12 shards

File: src/commands/developer/userdata.js (hunk 1 above)
Comment (klerikdust) [reply src/commands/developer/userdata.js:299]: Oh! yeah smart call. Its fine thenn

File: src/events/ready.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (klerikdust) [reply src/events/ready.js:14]: due to performance impact?

[the remark this answers is not in the document]
Comment (Moore2021) [reply src/events/ready.js:14]: mostly due to using broadcast eval, doesn't need to be called a bunch as its going to communicate across shards anyways, basically in my head if its called on every shard it becomes redundant code, as it should only be called once

**Comment by klerikdust** [comment]
Looking fine so far.
Does this involve any usage of redis? i didn't really dig deeper on whats inside. Just making sure
So that we know how to rollback the state in production if some issues occurring.

**Comment by Moore2021** [comment]
it is "using" redis but more so keeping track of guild ids, a roll back would require a backup of db

**Comment by klerikdust** [comment]
Gotcha, so both postres and redis rollback are needed for restoring the backup

**Comment by klerikdust** [comment]
Is this final? Its doable to merge

**Comment by Moore2021** [comment]
I believe it is doable, i can merge 3/7/2024 or 4/7/2024

**Pull request merged**